game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/game/selector/GameFileSelector.java

```diff
@@ -35,11 +36,10 @@
       fileDialog.setDirectory(ClientSetting.saveGamesFolderPath.getValueOrThrow().toString());
       fileDialog.setFilenameFilter((dir, name) -> GameDataFileUtils.isCandidateFileName(name));
       fileDialog.setVisible(true);
-      final String fileName = fileDialog.getFile();
-      final String dirName = fileDialog.getDirectory();
-      return Optional.ofNullable(fileName)
-          .map(name -> new File(dirName, fileName))
-          .map(this::mapFileResult);
+
+      // FileDialog.getFiles() always returns an array
+      // of 1 or 0 items
```

Comment:
The comment does not explain why we expect this to be true, why force a maintainer do API research when we can just say "this is the setting that makes this so".